File: dmn-test-cases/standard/tck/1.1/cl2/0009-invocation-arithmetic/translator/expected/java/dmn/type/TLoanImpl.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinition.ftl", "tLoan"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
public class TLoanImpl implements TLoan {
        private java.lang.Number amount;
        private java.lang.Number rate;
        private java.lang.Number term;

    public TLoanImpl() {
    }

    public TLoanImpl(java.lang.Number amount, java.lang.Number rate, java.lang.Number term) {
        this.setAmount(amount);
        this.setRate(rate);
        this.setTerm(term);
    }

    @com.fasterxml.jackson.annotation.JsonGetter("amount")
    public java.lang.Number getAmount() {
        return this.amount;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("amount")
    public void setAmount(java.lang.Number amount) {
        this.amount = amount;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("rate")
    public java.lang.Number getRate() {
        return this.rate;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("rate")
    public void setRate(java.lang.Number rate) {
        this.rate = rate;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("term")
    public java.lang.Number getTerm() {
        return this.term;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("term")
    public void setTerm(java.lang.Number term) {
        this.term = term;
    }

    @Override
    public boolean equals(Object o) {
        return equalTo(o);
    }

    @Override
    public int hashCode() {
        return hash();
    }

    @Override
    public String toString() {
        return asString();
    }
}

File: dmn-test-cases/standard/tck/1.1/cl3/0005-literal-invocation/translator/expected/kotlin/dmn/type/TLoanImpl.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinition.ftl", "tLoan"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
class TLoanImpl : TLoan {
    @get:com.fasterxml.jackson.annotation.JsonGetter("amount")
    @set:com.fasterxml.jackson.annotation.JsonGetter("amount")
    override var amount: kotlin.Number? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("rate")
    @set:com.fasterxml.jackson.annotation.JsonGetter("rate")
    override var rate: kotlin.Number? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("term")
    @set:com.fasterxml.jackson.annotation.JsonGetter("term")
    override var term: kotlin.Number? = null

    constructor() {
    }

    constructor (amount: kotlin.Number?, rate: kotlin.Number?, term: kotlin.Number?) {
        this.amount = amount
        this.rate = rate
        this.term = term
    }

    override fun equals(other: Any?): Boolean {
        return equalTo(other)
    }

    override fun hashCode(): Int {
        return hash()
    }

    @Override
    override fun toString(): String {
        return asString()
    }
}
